pkg/config/check.go

```diff
@@ -31,3 +40,67 @@ func checkCertificate() error {
 
 	return nil
 }
+
+const (
+	CertificateFile = "rootCA.pem"
+	KeyFile         = "rootCA-key.pem"
+)
+
+// Load loads the CA certificate and private key from the default filesystem locations.
+func Load() (*x509.Certificate, error) {
+	caRootPath, err := mkcertCARootPath()
+	if err != nil {
+		return nil, err
+	}
+
+	certPath := filepath.Join(caRootPath, CertificateFile)
+
+	cert, err := certificate.LoadCertificate(certPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse the CA certificate file (%s): %w", certPath, err)
+	}
+
+	return cert, nil
+}
+
+// AddToOS adds the CA to the operating system.
+func AddToOS(cert *x509.Certificate) error {
+	err := store.Add(cert)
+	if err != nil {
+		return fmt.Errorf("failed to add the CA to the operating system: %w", err)
+	}
+
+	return nil
+}
+
+// Get the mkcert CA root path depending on the OS.
+func mkcertCARootPath() (string, error) {
```

Comment:
I think this func is defined twice (here and in sni.go) maybe its posible to factorize ?

Reply:
It is. And check config shouldn't be a package. In my initial PR everything was done. But it was to big. I propose to live it like tha and then to clean the config/pkg.

Reply:
we will have to remove this function during merge conflict resolution (I did it locally for a dry run)